Test/Unit/Helper/ConfigTest.php

```diff
@@ -534,15 +514,266 @@ public function getIgnoredShippingAddressCoupons()
     /**
      * @test
      */
-    public function shouldTrackCheckoutFunnel()
+    public function getClientIp()
+    {
+        $request = $this->createMock(\Magento\Framework\App\Request\Http::class);
+        $this->setInaccessibleProperty($this->currentMock, '_request', $request);
+        $ip = '123.123.123.123';
+        $request->method('getServer')->with('HTTP_CLIENT_IP', false)->willReturn($ip);
+        $this->assertEquals($ip, $this->currentMock->getClientIp());
+    }
+
+    /**
+     * @test
+     */
+    public function getClientIp_false()
+    {
+        $request = $this->createMock(\Magento\Framework\App\Request\Http::class);
+        $this->setInaccessibleProperty($this->currentMock, '_request', $request);
+        $request->method('getServer')->withAnyParameters()->willReturn(false);
+        $this->assertEquals('', $this->currentMock->getClientIp());
+    }
+
+    /**
+     * @test
+     */
+    public function getIPWhitelistConfig()
+    {
+        $expected = '111.111.111.111,222.222.222.222';
```

Comment:
I think it will be useful to have a global variable tie together the IP's in this function and the one below

Reply:
done